snakemake/deployment/docker.py

```diff
@@ -0,0 +1,119 @@
+__author__ = "Johannes Köster"
+__copyright__ = "Copyright 2015-2020, Johannes Köster"
+__email__ = "johannes.koester@uni-due.de"
+__license__ = "MIT"
+
+import os
+import re
+import requests
+
+
+class DockerContainer:
+    """A Docker Container is a thin wrapper to a Docker Registry API to
+       primarily determine if a container exists to be used for a workflow
+    """
+
+    def __init__(self, name):
+        self.name = name
+        self.uri = None
+        if self.exists():
```

Comment:
This is fairly simple, but if a container exists, the DockerContainer will have a uri (that starts with docker:// to support how Snakemake currently expects to pull a singularity container). If the container doesn't exist, this returns None.